beacon_node/lighthouse_network/src/rpc/protocol.rs

```diff
@@ -194,6 +194,44 @@ pub enum Encoding {
     SSZSnappy,
 }
 
+/// The Encoding for all supported protocols is SSZSnappy.
+enum SupportedProtocol {
+    StatusV1,
+    GoodbyeV1,
+    BlocksByRangeV1,
+    BlocksByRangeV2,
+    BlocksByRootV1,
+    BlocksByRootV2,
+    PingV1,
+    MetaDataV1,
+    LightClientBootstrapV1,
+}
+
+impl SupportedProtocol {
+    fn currently_supported<T: EthSpec>(info: &RPCProtocol<T>) -> Vec<ProtocolId> {
```

Comment:
I was thinking in the future we could probably build a derive macro that builds this function for us and doesn't take the `info` parameter.